src/core/Chip/Chip.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { Component, ReactNode } from 'react';
+import classnames from 'classnames';
+import { default as styled } from 'styled-components';
+import { baseStyles } from './Chip.baseStyles';
+import { HtmlButton, HtmlSpan } from '../../reset';
+import { TokensProp, InternalTokensProp } from 'core/theme';
+import { withSuomifiDefaultProps } from '../theme/utils';
+import { Icon } from '../Icon/Icon';
+import { logger } from '../../utils/logger';
+import { VisuallyHidden } from '../../components/Visually-hidden/Visually-hidden';
+
+const baseClassName = 'fi-chip';
+const disabledClassName = `${baseClassName}--disabled`;
+const iconClassName = `${baseClassName}--icon`;
+const contentClassName = `${baseClassName}--content`;
+const removableClassName = `${baseClassName}--removable`;
+const buttonClassName = `${baseClassName}--button`;
+
+type ChipVariant = 'static' | 'default';
+
+interface InternalChipProps {
+  /** Chip element content */
+  children: ReactNode;
+  /** Custom class name for styling and customizing  */
+  className?: string;
+  /** Disable chip */
+  disabled?: boolean;
+  /**
+   * Event handler to execute when clicked
+   */
+  onClick?: () => void;
+  /**
+   * Show X-icon next to the content to mark the chip as removable
+   * @default false
+   */
+  removable?: boolean;
+  /** Aria-label attribute to let screen reader users know pressing the button will remove the chip/selection  */
+  actionLabel?: string;
+  /**
+   * default | static - use default for an interactive chip and static for a purely visual chip.
+   * @default default
+   */
+  variant?: ChipVariant;
+}
+
+export interface ChipProps extends InternalChipProps, TokensProp {}
+
+class DefaultChip extends Component<ChipProps> {
+  render() {
+    const {
+      className,
+      children,
+      onClick,
+      removable,
+      actionLabel,
+      variant,
+      disabled = false,
+      ...passProps
+    } = this.props;
+
+    if (removable && !actionLabel) {
+      logger.error(
+        'Provide actionLabel to communicate removability to screen readers',
+      );
+    }
+
+    if (variant === 'static') {
+      return (
+        <HtmlSpan
+          className={classnames(baseClassName, className, {
+            [disabledClassName]: !!disabled,
+            [removableClassName]: !!removable,
+          })}
+          {...passProps}
+        >
+          <HtmlSpan className={contentClassName}>{children}</HtmlSpan>
```

Comment:
Is this additional span necessary?

Reply:
This one might actually be redundant. I'll look into it.

Reply:
Removing this would require some duplicate code and restructuring in the styles, so this might be a cleaner way to control the size and overflow.

Reply:
Sounds reasonable, so let's keep it as it is even if not immediately obvious why it's there.